cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/datastore/DatastoreSchemaTests.java

```diff
@@ -251,4 +254,135 @@ private static File f(String filename) {
         return TestUtils.loadFixture(filename);
     }
 
+    /**
+     * <p>
+     * Tests the migration of an existing datastore "corrupted" by the issues 329 and 326. Some
+     * attachments have also been added to exercise that code. The database contains many documents
+     * that have multiple identical revisions, some of which are deleted (and because of 326) have
+     * no current winner. Others have multiple identical revisions, but are otherwise normal.
+     * </p>
+     * <p>
+     * The special case documents are:
+     * </p>
+     * <ul>
+     * <li>
+     * 300 d834ca038de24bf0ac9f708fcdb63e21 1-abea4faccec4430685dd2065f28a7dd9:
+     * sequence 594 (lowest) has 3 identical attachment entries, seq 847 has no attachments
+     * </li>
+     * <li>
+     * 307 a2359c3503e34c008ec448834583e482 1-b83f5c820588498ca2228282237d4f90:
+     * sequence 1051 (highest) has 2 identical attachment entries, seq 606 has no attachments
+     * </li>
+     * <li>
+     * 308 badad1b3842e4056b013587b49c93308 1-aaf13ff7c5dc44ba8626f684fb73b9d3:
+     * sequence 607 (lowest) 1050 (highest) each have one copy of each of two different attachments
+     * </li>
+     * </ul>
+     *
```

Comment:
I think a diagram of these special cases would make it easier to understand where the attachments sit in the tree